### DISCUSSION TITLE
Experiment cycle time --> cycle number

### DISCUSSION THREAD

**Discussion by presvetli** [pybamm-team/PyBaMM#2007] [Q&A]
For certain tests, perform 500 cycles (discharge and charge)*500. I understand how to calculate cycle time length from a solution and then plot graphs of cycle number vs. some variable. 
I need cycle number information for some parameter function during calculation. Is it possible to retrieve or set a variable for Cycle number based on cycle time during calculation?
Do you have any suggestions for how to tackle this problem when the parameters are dependent on the cycle number (energy throughput)?

**Comment by valentinsulzer** [comment]
Just so I understand correctly, are you wanting to have a parameter `a` vary as a function of cycle number `n`, e.g. `a = 1 - n/1000`? Or asking about how to post-process to plot how a parameter changes with cycle number?

**Reply by presvetli** [reply]
I want to have a parameter a vary as a function of cycle number n, e.g. a = 1 - n/1000.

**Reply by valentinsulzer** [reply]
This isn't really recommended as it's not very physically justified. e.g. how do you adapt this if instead of full cycles you have 50% DOD? Also, the solver might not like if you have too large discontinuities in the parameter between cycles. It's better to set up an ODE for your degradation variable, for example SEI thickness increasing proportionally to the integral of SEI current.

Having said that, one way you could do it is to make the parameter an "input", then at each cycle update the input, something like
```python
import pybamm
import matplotlib.pyplot as plt

model = pybamm.lithium_ion.SPM()
experiment = pybamm.Experiment(
    [
        (
            "Discharge at C/5 for 10 hours or until 3.3 V",
            "Rest for 1 hour",
            "Charge at 1 A until 4.1 V",
            "Hold at 4.1 V until 10 mA",
            "Rest for 1 hour",
        ),
    ]
)
model = pybamm.lithium_ion.DFN()
param = pybamm.ParameterValues("Chen2020")
eps_n_init = param["Negative electrode active material volume fraction"]
param["Negative electrode active material volume fraction"] = pybamm.InputParameter(
    "eps_n"
)
sim = pybamm.Simulation(
    model,
    experiment=experiment,
    solver=pybamm.CasadiSolver("fast with events"),
    parameter_values=param,
)

sol = None
for i in range(5):
    print(i)
    eps_n = eps_n_init * (1 - i / 100)
    sol = sim.solve(inputs={"eps_n": eps_n}, starting_solution=sol)

fig, ax = plt.subplots()
for cycle in sol.cycles:
    t = cycle["Time [s]"].data
    ax.plot(t - t[0], cycle["Terminal voltage [V]"].data)

sol.plot(["X-averaged negative electrode active material volume fraction"])
```

**Reply by presvetli** [reply]
Thank you very much for your support. It is working well now.
